Tools/LambdaTestTool-v2/src/Amazon.Lambda.TestTool/Extensions/HttpContextExtensions.cs

```diff
@@ -0,0 +1,106 @@
+
+        var httpApiV2ProxyRequest = new APIGatewayHttpApiV2ProxyRequest
+        {
+            RouteKey = $"{request.Method} {apiGatewayRouteConfig.Path}",
+            RawPath = request.Path,
+            RawQueryString = request.QueryString.Value,
+            Cookies = request.Cookies.Select(c => $"{c.Key}={c.Value}").ToArray(),
+            Headers = headers,
+            QueryStringParameters = queryStringParameters,
+            PathParameters = pathParameters ?? new Dictionary<string, string>(),
+            Body = HttpRequestUtility.ReadRequestBody(request),
+            IsBase64Encoded = false,
+            RequestContext = new ProxyRequestContext
+            {
+                Http = new HttpDescription
+                {
+                    Method = request.Method,
+                    Path = request.Path,
+                    Protocol = request.Protocol
+                },
+                RouteKey = $"{request.Method} {apiGatewayRouteConfig.Path}"
+            },
+            Version = "2.0"
+        };
+
+        if (HttpRequestUtility.IsBinaryContent(request.ContentType))
+        {
+            httpApiV2ProxyRequest.Body = Convert.ToBase64String(Encoding.UTF8.GetBytes(httpApiV2ProxyRequest.Body));
+            httpApiV2ProxyRequest.IsBase64Encoded = true;
+        }
+
+        return httpApiV2ProxyRequest;
+    }
+
+    /// <summary>
+    /// Translates an <see cref="HttpContext"/> to an <see cref="APIGatewayProxyRequest"/>.
+    /// </summary>
+    /// <param name="context">The <see cref="HttpContext"/> to be translated.</param>
+    /// <param name="apiGatewayRouteConfig">The configuration of the API Gateway route, including the HTTP method, path, and other metadata.</param>
+    /// <returns>An <see cref="APIGatewayProxyRequest"/> object representing the translated request.</returns>
+    public static APIGatewayProxyRequest ToApiGatewayRequest(
+        this HttpContext context,
+        ApiGatewayRouteConfig apiGatewayRouteConfig)
+    {
+        var request = context.Request;
+
+        var pathParameters = RouteTemplateUtility.ExtractPathParameters(apiGatewayRouteConfig.Path, request.Path);
+
+        var (headers, multiValueHeaders) = HttpRequestUtility.ExtractHeaders(request.Headers);
+        var (queryStringParameters, multiValueQueryStringParameters) = HttpRequestUtility.ExtractQueryStringParameters(request.Query);
+
+        var proxyRequest = new APIGatewayProxyRequest
+        {
+            Resource = apiGatewayRouteConfig.Path,
+            Path = HttpUtility.UrlEncode(request.Path),
+            HttpMethod = request.Method,
+            Headers = headers,
+            MultiValueHeaders = multiValueHeaders,
+            QueryStringParameters = queryStringParameters,
+            MultiValueQueryStringParameters = multiValueQueryStringParameters,
+            PathParameters = pathParameters ?? new Dictionary<string, string>(),
```

Comment:
Why are we initializing this to an empty collection if we don't have any. Is this what API Gateway does?

Reply:
good catch. i just did a test
`Invoke-WebRequest -Uri https://myapi-gateway-url.amazonaws.com/testfunction -Method GET`
where it echos the api gateway request object and i see the value is `null` when there are no path parameters. 
```
pathParameters: null,
  stageVariables: null,
  body: null,
  isBase64Encoded: false
}
```

i will update the code

Reply:
ive updated in fecf91d to handle this